Nodejs/Product/TestAdapter/TestFilesUpdateWatcher.cs

```diff
@@ -43,84 +39,71 @@ public bool AddDirectoryWatch(string path)
         {
             ValidateArg.NotNull(path, "path");
 
-            if (!string.IsNullOrEmpty(path))
+            if (!this.watchedFiles.ContainsKey(path) && ErrorHandler.Succeeded(this.fileWatcher.AdviseDirChange(path, VSConstants.S_OK, this, out uint cookie)))
             {
-                if (!_fileWatchers.ContainsKey(path) && Directory.Exists(path))
-                {
-                    var watcher = new FileSystemWatcher(path);
-                    _fileWatchers[path] = watcher;
-
-                    watcher.IncludeSubdirectories = true;
-                    watcher.Changed += OnChanged;
-                    watcher.Renamed += OnRenamed;
-                    watcher.EnableRaisingEvents = true;
-                    return true;
-                }
+                this.watchedFiles.Add(path, cookie);
+                return true;
             }
             return false;
         }
 
-        public void RemoveWatch(string path)
+        public bool RemoveFileWatch(string path)
         {
             ValidateArg.NotNull(path, "path");
 
-            if (!string.IsNullOrEmpty(path))
+            if (this.watchedFiles.TryGetValue(path, out uint cookie))
             {
-                FileSystemWatcher watcher;
-
-                if (_fileWatchers.TryGetValue(path, out watcher))
-                {
-                    watcher.EnableRaisingEvents = false;
-
-                    _fileWatchers.Remove(path);
-
-                    watcher.Changed -= OnChanged;
-                    watcher.Dispose();
-                }
+                this.watchedFiles.Remove(path);
+                return ErrorHandler.Succeeded(this.fileWatcher.UnadviseFileChange(cookie));
             }
+            return false;
         }
 
-        private void OnChanged(object sender, FileSystemEventArgs e)
+        public bool RemoveFolderWatch(string path)
```

Comment:
"RemoveDirectoryWatch" for symmetry with "AddDirectoryWatch"

Reply:
I'm going with `AddFolderWatch`